From a7b805e86385db6840625a3e45f7dc6ca81eba28 Mon Sep 17 00:00:00 2001
From: Tamal <tasaha@habana.ai>
Date: Fri, 04 Oct 2024 13:18:43 +0300
Subject: [PATCH] [SW-190823] g3: lzop compress engine bist pattern

This patch creates compressed engine bist pattern
using lzop. This data will be decompressed in preboot
and used for engine bist execution.

TODO: Tried to compressed the data using lzop and
stored in SPI. Uncompressed it in preboot using
lzop_decompress. Compressed data stored in spi
and loaded data from spi to memory using dma,
matches in preboot. However uncompressed data
using lzop_decompress does not match with original
data.

Change-Id: I427e3cd6917954bd168613a26b8806959eea8732
Signed-off-by: Tamal <tasaha@habana.ai>
---

diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index 1b3de94..d7c1c72 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -232,10 +232,10 @@
 	local CACHE_MBIST_DATA_BINFILE=$1
 
 	xxd -r -p $DIE_ENGINE_MBIST_DATA_TEXT_FILE > ${TEMP_ENGINE_MBIST_TABLE_BINFILE}
-	#${LZOP} -f ${TEMP_ENGINE_MBIST_TABLE_BINFILE} -o ${ENGINE_MBIST_DATA_BINFILE}
+	${LZOP} -f ${TEMP_ENGINE_MBIST_TABLE_BINFILE} -o ${ENGINE_MBIST_DATA_BINFILE}
 	#${LZOP} -d ${ENGINE_MBIST_DATA_BINFILE} -o ${TEMP2_ENGINE_MBIST_TABLE_BINFILE}
 
-	xxd -r -p $DIE_ENGINE_MBIST_DATA_TEXT_FILE > ${ENGINE_MBIST_DATA_BINFILE}
+	#xxd -r -p $DIE_ENGINE_MBIST_DATA_TEXT_FILE > ${ENGINE_MBIST_DATA_BINFILE}
 	xxd -r -p $DIE_SRAM_MBIST_DATA_TEXT_FILE > ${CACHE_MBIST_DATA_BINFILE}
 
 	hl_pack_bin ${CACHE_MBIST_DATA_BINFILE} 8k 0 0 0
